sync-pingdom-ec2-security-groups.py

```diff
@@ -99,6 +100,7 @@ def create_permission(self, ip):
         return Permission(self.protocol, '{0}/32'.format(ip), self.from_port, self.to_port)
 
     def configure_permissions(self, permissions):
+        boto3.setup_default_session(profile_name=self.profile)
```

Comment:
This method modifies a global default. Having looked at alternatives, I think we should pass around a [Session](https://boto3.readthedocs.io/en/latest/reference/core/session.html) object rather than the individual `region` and `profile` parameters.